kernel: pipes: remove simple dead function k_pipe_block_put

s-kelvin opened the pull request:
Removed k_pipe_block_put and static functions only related to it. After all the old usage of k_mem_block has been replaced by k_heap in [24358](https://github.com/zephyrproject-rtos/zephyr/issues/24358), k_pipe_block_put which still takes a deprecated k_mem_block as the argument becomes dead code. All APIs that hook it from kernel.h have been confirmed to be removed. 
Since an asynchronous message descriptor is only allocated in k_pipe_block_put, static functions for pipe_async are removed as well.

Signed-off-by: Shihao Shen <shihao.shen@intel.com>

LeiW000 commented:
For the codes, it looks good to me. We still need to update the document on zephyr website to remove the API.

One thing I would like to confirm that all tests cases under tests/kernel/pips have been run to make sure that there is no regression bug?

andyross reviewed (approved):
Looks great, thanks.

LeiW000 reviewed (approved):
Please remember to update the doc on Zephyr project website to remove the API.

LeiW000 commented:
@andyross @nashif , Can it be merged now?

nashif reviewed (changes requested):
This can't be removed now, only after 2.5 is released. Also, I do not see you removing this from kernel.h

see https://docs.zephyrproject.org/latest/releases/release-notes-2.4.html#deprecated-in-this-release

SecureSheII committed:
kernel: pipes: remove simple dead function k_pipe_block_put

Removed k_pipe_block_put and static functions only related to it.
After all the old usage of k_mem_block has been replaced by k_heap,
k_pipe_block_put still taking a deprecated k_mem_block as argument
becomes dead code. All APIs that hook it from kernel.h have been
confirmed to be removed. Since an asynchronous message descriptor
is only allocated in k_pipe_block_put, static functions for pipe_
async are removed as well.

Signed-off-by: Shihao Shen <shihao.shen@intel.com>

s-kelvin commented:
Removed in kernel.h as well. Will wait for merge till the release of 2.5. Thank you.

s-kelvin commented:
Can this be merged now? @nashif

LeiW000 commented:
2.5 release is already out. I think the PR can be merged now.

LeiW000 commented:
@nashif, Is it OK to merge it now? 2.5 release is already out.

nashif commented:
We can't remove an API like this from kernel.h without deprecating it first, someone might be using it? Add a deprecation first, then it can be remove later.

s-kelvin commented:
@nashif Actually k_pipe_block_put has been marked to be deprecated in 2.4 per https://docs.zephyrproject.org/latest/releases/release-notes-2.4.html#deprecated-in-this-release. And because now k_mem_pool API has been removed entirely in 2.5, should we remove k_pipe_block_put as well? It's a simple dead function that no other APIs in kernel.h hook. 
If we are not going to remove it, I could just `__deprecated` it now and wait for 2.6 to remove it. Also I am going to exclude the lines in pipes.c that are related to k_pipe_block_put from code coverage. Let me know if this is what's expected to do and I will resubmit my commit.

LeiW000 commented:
I just took look at the 2.4 release note again. It says that k_pipe_block_put() will be removed in 2.6. That is what Kelvin has done in this PR. Now, the 2.5 release is already out, and we are in the development phase of 2.6. I don't see anything wrong with this PR. @nashif, what do you think?
